### PR TITLE
Hide quote related exceptions from users

### PR DESCRIPTION
**Pull request by nicolasstucki**
Users should never throw or catch these exceptions.

### CODE REVIEW

File: library/src/scala/internal/quoted/ScopeException.scala [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package scala.quoted
+package scala.internal.quoted
```
Comment (liufengyun) [inline library/src/scala/internal/quoted/ScopeException.scala:1]: What about `private`? Or use `scala.quoted.internal`?

Comment (nicolasstucki) [reply library/src/scala/internal/quoted/ScopeException.scala:1]: There is one use in the compiler. I will try to remove it and make it private.

Comment (nicolasstucki) [reply library/src/scala/internal/quoted/ScopeException.scala:1]: I made it private

**Review by liufengyun — APPROVED** [review]
LGTM

**Pull request merged**